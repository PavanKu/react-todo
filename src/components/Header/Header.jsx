import React from "react";
import PropTypes from "prop-types";
import StyledHeader from "./Header.styled";

export const APPLICATION_DEFAULT_TITLE = "todos";

function Header({ title = APPLICATION_DEFAULT_TITLE }) {
  return <StyledHeader className="appTitle">{title}</StyledHeader>;
}

Header.propTypes = {
  title: PropTypes.string,
};

export default Header;
