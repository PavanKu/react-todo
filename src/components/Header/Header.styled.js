import styled from "styled-components";

const StyledHeader = styled.h1`
  font-size: 100px;
  text-align: center;
  color: rgba(175, 47, 47, 0.15);
  text-rendering: optimizeLegibility;
  -moz-text-rendering: optimizeLegibility;
  font-weight: 100;
`;

export default StyledHeader;
