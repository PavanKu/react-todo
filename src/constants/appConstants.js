const TAB_OPTIONS = {
  ALL: "All",
  ACTIVE: "Active",
  COMPLETED: "Completed",
};

export { TAB_OPTIONS };
