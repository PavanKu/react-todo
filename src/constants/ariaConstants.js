const ARIA_ROLE_HEADING = "heading";
const ARIA_ROLE_TEXTBOX = "textbox";
const ARIA_ROLE_CHECKBOX = "checkbox";
const ARIA_ROLE_BUTTON = "button";
const ARIA_ROLE_TABLIST = "tablist";
const ARIA_ROLE_TAB = "tab";

const ARIA_TAB_SELECTED_ATTRIBUTE = "aria-selected";

export {
  ARIA_ROLE_HEADING,
  ARIA_ROLE_TEXTBOX,
  ARIA_ROLE_CHECKBOX,
  ARIA_ROLE_BUTTON,
  ARIA_ROLE_TABLIST,
  ARIA_ROLE_TAB,
  ARIA_TAB_SELECTED_ATTRIBUTE,
};
