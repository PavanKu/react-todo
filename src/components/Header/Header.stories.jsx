import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import Header, { APPLICATION_DEFAULT_TITLE } from "./Header";

const DefaultHeader = () => <Header />;
const Playground = () => (
  <Header title={text("title", APPLICATION_DEFAULT_TITLE)} />
);

storiesOf("Header", module)
  .addDecorator(withKnobs)
  .add("Default Header", DefaultHeader)
  .add("Playground", Playground);
