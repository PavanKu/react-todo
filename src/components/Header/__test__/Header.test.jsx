import React from "react";
import { cleanup, render } from "@testing-library/react";
import Header, { APPLICATION_DEFAULT_TITLE } from "../Header";
import { ARIA_ROLE_HEADING } from "../../../constants/ariaConstants";

describe("<Header />", () => {
  afterEach(cleanup);

  it("should render without crashing", () => {
    render(<Header />);
  });

  it("should render with default title", () => {
    const { getByRole } = render(<Header />);
    const elementByRole = getByRole(ARIA_ROLE_HEADING);

    expect(elementByRole).toHaveTextContent(APPLICATION_DEFAULT_TITLE);
  });

  it("should render as per props", () => {
    const APP_TITLE = "Hello World";

    const { getByRole } = render(<Header title={APP_TITLE} />);
    const elementByRole = getByRole(ARIA_ROLE_HEADING);

    expect(elementByRole).toHaveTextContent(APP_TITLE);
  });

  it("should match snapshot", () => {
    const { container } = render(<Header title={APPLICATION_DEFAULT_TITLE} />);
    expect(container).toMatchSnapshot();
  });
});
