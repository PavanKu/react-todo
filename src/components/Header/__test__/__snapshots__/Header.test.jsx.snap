// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <h1
    class="sc-AxjAm hsvQWr appTitle"
  >
    todos
  </h1>
</div>
`;
